python/tests/test_bins.py

```diff
@@ -45,6 +45,43 @@ def test_bins():
     assert sc.is_equal(var['y', 1].value, data['x', 2:4])
 
 
+def test_bins_view():
+    col = sc.Variable(dims=['event'], values=[1, 2, 3, 4])
+    table = sc.DataArray(data=col,
+                         coords={'time': col * 2.2},
+                         attrs={'attr': col * 3.3},
+                         masks={'mask': col == col})
+    begin = sc.Variable(dims=['y'], values=[0, 2], dtype=sc.dtype.int64)
+    end = sc.Variable(dims=['y'], values=[2, 4], dtype=sc.dtype.int64)
+    var = sc.bins(begin=begin, end=end, dim='event', data=table)
+    assert 'time' in var.bins.coords
+    assert 'time' in var.bins.meta
+    assert 'attr' in var.bins.meta
+    assert 'attr' in var.bins.attrs
+    assert 'mask' in var.bins.masks
+    with pytest.raises(RuntimeError):
+        var.bins.coords['time2'] = col  # col is not binned
+
+    def check(a, b):
+        # sc.is_equal does not work for us here since we have owning and
+        # non-owning views which currently never compare equal.
+        assert sc.all(a == b).bins.sum().value
```

Comment:
Does not catch cases where only some elements are different as long as at least one matches. This should work: `sc.all(sc.all(a==b).values).value`